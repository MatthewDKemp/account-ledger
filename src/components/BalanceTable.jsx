import React from "react";
import CurrencyFormat from "react-currency-format";
import Table from "react-bootstrap/Table";
import { useAccountLedgerContainer } from "../containers/accountLedger.container";

export function BalanceTable() {
  const { overdraftLimit, balance } = useAccountLedgerContainer();
  return (
    <>
      <b>
        <u>Account Summary</u>
      </b>
      <Table
        bordered
        style={{ width: "80%", marginLeft: "auto", marginRight: "auto" }}
      >
        <thead>
          <tr>
            <th
              style={{
                width: "33.33%",
              }}
            >
              Overdraft{" "}
            </th>
            <th
              style={{
                width: "33.33%",
              }}
            >
              Current Balance{" "}
            </th>
            <th
              style={{
                width: "33.33%",
              }}
            >
              Available Balance{" "}
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <CurrencyFormat
                value={Math.abs(overdraftLimit)}
                displayType={"text"}
                thousandSeparator={true}
                prefix={"£"}
              />
            </td>
            <td>
              {" "}
              <CurrencyFormat
                value={balance}
                displayType={"text"}
                thousandSeparator={true}
                prefix={"£"}
              />
            </td>
            <td>
              {" "}
              <CurrencyFormat
                value={balance + Math.abs(overdraftLimit)}
                displayType={"text"}
                thousandSeparator={true}
                prefix={"£"}
              />
            </td>
          </tr>
        </tbody>
      </Table>
    </>
  );
}
