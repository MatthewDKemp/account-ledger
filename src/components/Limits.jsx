import React from "react";
import CurrencyFormat from "react-currency-format";
import { useAccountLedgerContainer } from "../containers/accountLedger.container";

export function Limits() {
  const { overdraftLimit, highBalancePrompt } = useAccountLedgerContainer();
  return (
    <>
      <b>
        <u>Account Limits</u>
      </b>
      <div>
        <label>Agreed overdraft:</label>{" "}
        <CurrencyFormat
          value={overdraftLimit}
          displayType={"text"}
          thousandSeparator={true}
          prefix={"£"}
        />
        <br></br>
        <label>High balance prompt:</label>{" "}
        <CurrencyFormat
          value={highBalancePrompt}
          displayType={"text"}
          thousandSeparator={true}
          prefix={"£"}
        />
      </div>
    </>
  );
}
