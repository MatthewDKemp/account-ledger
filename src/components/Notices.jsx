import React from "react";
import { useAccountLedgerContainer } from "../containers/accountLedger.container";

export function Notices() {
  const { displayNotice, inAgreedOverdraft, highBalance } =
    useAccountLedgerContainer();
  return (
    <>
      <div
        style={{
          width: "80%",
          marginLeft: "auto",
          marginRight: "auto",
          marginBottom: "15px",
        }}
      >
        {displayNotice && (
          <>
            <br></br>
            <b>
              <u>Account Notices</u>
            </b>
          </>
        )}
        {inAgreedOverdraft && <p>You are in your agreed overdraft</p>}
        {highBalance && (
          <p>
            Your balance is at least Â£4,000, consider saving some of this for a
            rainy day :)
          </p>
        )}
      </div>
    </>
  );
}
