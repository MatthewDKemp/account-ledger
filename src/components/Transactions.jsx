import React from "react";
import CurrencyFormat from "react-currency-format";
import Table from "react-bootstrap/Table";
import { useAccountLedgerContainer } from "../containers/accountLedger.container";

export function Transactions() {
  const {
    invalidDate,
    dateError,
    invalidAmount,
    amountError,
    balanceArray,
    recordTransaction,
  } = useAccountLedgerContainer();
  return (
    <>
      <div>
        <b>
          <u>Log New Transaction</u>
        </b>

        <div>
          <div>
            <b>Transaction date:</b>
            <br></br>
            <input
              id="dateInput"
              type="date"
              style={{ height: "50px", marginBottom: "15px" }}
            />
            {invalidDate && (
              <p
                id="dateInputError"
                style={{
                  backgroundColor: "yellow",
                  width: "80%",
                  marginBottom: "15px",
                  marginLeft: "auto",
                  marginRight: "auto",
                }}
              >
                {dateError}
              </p>
            )}
          </div>
          <div>
            <b>Transaction amount (£):</b>
            <br></br>
            <input
              id="amountInput"
              type="number"
              autoComplete="off"
              style={{ height: "50px", marginBottom: "15px" }}
            />
            {invalidAmount && (
              <p
                id="amountInputError"
                style={{
                  backgroundColor: "yellow",
                  width: "80%",
                  marginBottom: "15px",
                  marginLeft: "auto",
                  marginRight: "auto",
                }}
              >
                {amountError}
              </p>
            )}
          </div>

          <button type="button" onClick={recordTransaction}>
            Record transaction
          </button>
        </div>
      </div>
      <br></br>
      <b>
        <u>Transaction List</u>
      </b>
      <Table
        bordered
        style={{ width: "80%", marginLeft: "auto", marginRight: "auto" }}
      >
        <thead>
          <tr>
            <th
              style={{
                width: "33.33%",
              }}
            >
              Date
            </th>
            <th
              style={{
                width: "33.33%",
              }}
            >
              Amount
            </th>
            <th
              style={{
                width: "33.33%",
              }}
            >
              Balance
            </th>
          </tr>
        </thead>
        <tbody>
          {balanceArray.map((item) => {
            return (
              <tr key={item.id}>
                <td>{item.date}</td>
                <td>
                  <CurrencyFormat
                    value={item.amount}
                    displayType={"text"}
                    thousandSeparator={true}
                    prefix={"£"}
                  />
                </td>
                <td>
                  <CurrencyFormat
                    value={item.balance}
                    displayType={"text"}
                    thousandSeparator={true}
                    prefix={"£"}
                  />
                </td>
              </tr>
            );
          })}
        </tbody>
      </Table>
    </>
  );
}
