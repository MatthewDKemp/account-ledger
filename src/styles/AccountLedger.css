.MainContent {
  background-color: white;
  align-content: center;
  text-align: center;
  justify-content: center;
  width: 50%;
}
