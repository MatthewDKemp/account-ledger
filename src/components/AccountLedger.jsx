import React from "react";
import "../styles/AccountLedger.css";
import { BalanceTable } from "./BalanceTable";
import { Limits } from "./Limits";
import { Notices } from "./Notices";
import { Transactions } from "./Transactions";

export function AccountLedger() {
  // Set up initial transactions to be displayed on initial render

  return (
    <>
      <div className="MainContent">
        <BalanceTable />
        <Limits />
        <Notices />
        <Transactions />
      </div>
    </>
  );
}
