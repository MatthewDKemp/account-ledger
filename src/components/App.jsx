import React from "react";
import { AccountLedger } from "./AccountLedger";
import "../styles/App.css";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <AccountLedger />
      </header>
    </div>
  );
}

export default App;
