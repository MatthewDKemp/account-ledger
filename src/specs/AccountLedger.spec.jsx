import React from "react";
import { shallow } from "enzyme";
import { AccountLedger } from "../components/AccountLedger";

describe("Account Ledger Component Test", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const shallowComponent = () => shallow(<AccountLedger />);

  it("should render Counter component", () => {
    const component = shallowComponent();
    expect(component).toMatchSnapshot();
  });
});
