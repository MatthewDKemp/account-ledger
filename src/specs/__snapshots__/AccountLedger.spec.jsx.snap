// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Ledger Component Test should render Counter component 1`] = `ShallowWrapper {}`;
